### ISSUE TITLE
Database manager for driver is not registered.

### ISSUE THREAD

**Issue by greggunner** [stancl/tenancy#841]
### Bug description

We just deployed an installation of our application and getting the error "Database manager for driver is not registered." when trying to access the tenant website. This is thrown at vendor/stancl/tenancy/src/DatabaseConfig.php:156.

On further investigation it seems that the database config file isn't being fully loaded or similar.

Our config/database.php file provides a config for the tenant db settings. "database.connections.tenant".

But when line vendor/stancl/tenancy/src/DatabaseConfig.php:151 runs with `$driver = config("database.connections.{$this->getTemplateConnectionName()}.driver");`, which the config name resolves to "database.connections.tenant.driver", the value of $driver is null.

### Steps to reproduce

Project specific. So willing to provide details as needed.

### Expected behavior

$driver is meant to contain "pgsql" based on the database settings in our config file.

### Laravel version

8.69.0

### stancl/tenancy version

3.5.0

**Comment by stancl** [comment]
Could this be related to #840?

**Comment by stancl** [comment]
Ah no, that's Laravel 9. Did this issue start to happen at a specific version of Tenancy?

**Comment by greggunner** [comment]
Thanks Samuel... can't comment on a specific version. Only that it is happening now with the deploy and we are on 3.4.5.

**Comment by stancl** [comment]
I can't really do much about this if I don't know which version introduced it.

If downgrading doesn't fix it, the issue is likely on your end.

**Comment by greggunner** [comment]
Ok. Perhaps you could point me in a direction to look at in the code? What I don't understand is how the config value is null when it is in the config files, and on my local version it works as expected.

**Comment by stancl** [comment]
I think it'd be best to handle this as a support question on our Discord. Try to provide the reproduction steps or describe your setup.

**Comment by greggunner** [comment]
Hi Stancl. I finally got time to investigate this further and found that when I upgrade from 3.4.x to 3.5.0, the issue starts.

Our setup:

- We have a single central database with a database per tenant.
- Tenant config: to setup an additional config details specific for each tenant using `\Stancl\Tenancy\Features\TenantConfig::$storageToConfigMap`

I tried disabling the TenantConfig to see if that caused the issue and it didn't.

**Comment by greggunner** [comment]
@stancl can we get this bug re-opened please.

**Comment by stancl** [comment]
> 3.4.x to 3.5.0

Which exact version introduced this?

**Comment by greggunner** [comment]
3.5.0. 

I tested 3.4.5 and 3.4.6, and then 3.5.0. It only started on 3.5.0.

**Comment by greggunner** [comment]
Just a note: the error message says "Database manager for driver is not registered." But obviously it's missing a word as the driver name is null. Should be something like "Database manager for driver [name] is not registered."

**Comment by stancl** [comment]
So the bug should be introduced in this diff right: https://github.com/archtechx/tenancy/compare/v3.4.6...v3.5.0

**Comment by stancl** [comment]
iirc the main change in that version is that we don't keep `tenant` connections alive after the tenant context is left (which is the correct behavior; previously they were kept open). The implications are that:
- you can't return tenant models from `$tenant->run()` and use them like normal models. You should instead return the *final* data you want
- you can't call `toArray()` / `toCollection()` on those models since that tries to serialize timestamps using the DB connection's format. The connection doesn't exist anymore, so 
    - I attempted to fix this in a PR to laravel/framework, but couldn't finish it just yet https://github.com/laravel/framework/pull/40515
- some packages like Telescope attempt to call `toArray()` on models passed to views

So in general, there are now stronger boundaries between the central & tenant context and data (or especially DB connections) aren't allowed to leak into the other context.

**Comment by davidrushton** [comment]
@stancl @greggunner I also ran into this issue when updating the package from 3.4.x to 3.5.x.

My fix was to update my `config/tenancy.php` and change:

```php
        'template_tenant_connection' => 'tenant',
```

to

```php
        'template_tenant_connection' => 'tenant_template',
```

(and add that new `tenant_template` connection in my `config/database.php` file). 

I also had to keep my old `tenant` database connection as that was the default and I was getting `InvalidArgumentException: Database connection [tenant] not configured.` as my AppServiceProvider has 

```php
Schema::defaultStringLength(191);
```

which was trying to do things to the default connection.

Hope that helps?
